hbase/src/main/java/com/yahoo/ycsb/db/HBaseClient10.java

```diff
@@ -246,17 +266,22 @@ public int read(String table, String key, Set<String> fields, HashMap<String,Byt
             //do nothing for now...need to understand HBase concurrency model better
             return ServerError;
         }
-
-        if (r.isEmpty()) {
+        if (multiGetBatchSize == 0) {
+          if (r[0].isEmpty()) {
             return NoMatchingRecord;
+          }
         }
-        for (Cell c : r.listCells()) {
-            result.put(Bytes.toString(CellUtil.cloneQualifier(c)),
-                    new ByteArrayByteIterator(CellUtil.cloneValue(c)));
-            if (_debug) {
-                System.out.println("Result for field: "+Bytes.toString(CellUtil.cloneQualifier(c))+
-                        " is: "+Bytes.toString(CellUtil.cloneValue(c)));
+        for (Result res : r) {
+          if(!res.isEmpty()) {
```

Comment:
You mean with this change
{code}
if (multiGetBatchSize == multiGets.size()) {
{code}
I get it now. I thought the number of gets will be good enough to match it. I will update the code as I had it in my internal patch. I initialized the Result[] to null and for the non multiGet case I initialized the array locally.